### PR TITLE
feat: Introduce Oauth integration for distributor and Oauth enhancements for CLI

### LINKED ISSUE #6728: Distributor: add support for client credential oauth flow for remote execution plane
The distributor shall support configurable support for performing the client credential (machine to machine) OAuth flow when calling the public Keptn API.

The way to configure the relevant information shall be possible via environment variable:

(1) `OAUTH_CLIENT_ID` - the Oauth client ID, `string default=""`
(2) `OAUTH_CLIENT_SECRET` - the Oauth client secret  `string default=""`
(3) `OAUTH_SCOPES` - the Oauth Scopes used during the oauth flow  `string default=""`
(4) `OAUTH_TOKEN_URL` - the URL to obtain the token from `string default=""`

All of these must be set in order to activate the OAuth flow.

### Acceptance criteria
```
Given the distributor is not running
When the distributor is started on a remote execution plane with (1), (2), (3). (4) set, and (5) not set
Then the distributor uses a HTTP client that supports Oauth Client Credentials flow and does not contain the Keptn API token
```

```
Given the distributor is not running
When the distributor is started on a remote execution plane with (1), (2), (3). (4) and (5) set
Then the distributor uses a HTTP client that supports Oauth Client Credentials flow and also contain the Keptn API token
```

```
Given the distributor is not running
When the distributor is started on a remote execution plane with one of (1), (2), (3) or (4) not set
Then the distributor shall use a HTTP client which is not configured for Oauth and does contain the KEPTN API token.
```

```
Given the distributor is not running
When the distributor is started on the control plane
Then the distributor shall ignore the values of (1), (2), (3), (4), and (5)
```

### PR DESCRIPTION
**Pull request by warber**
closes #6728 

PR contains: 
* changes according to #6728 
* rename of cli oauth flags to have prefix `oauth` instead of `sso`
* bug fixes in cli
* ability to setup oauth for distributor in helm installer

Integration Tests Run: https://github.com/keptn/keptn/actions/runs/1851816917

### CODE REVIEW

**Review by bacherfl — APPROVED** [review]

File: jmeter-service/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -147,6 +147,16 @@ spec:
                 fieldRef:
                   fieldPath: metadata.labels['app.kubernetes.io/name']
             {{- end }}
+            - name: OAUTH_CLIENT_ID
+              value: "{{ (((.Values.distributor).config).oauth).clientID }}"
+            - name: OAUTH_CLIENT_SECRET
```
Comment (mowies) [inline jmeter-service/chart/templates/deployment.yaml:152]: I think the same applies here as for the bridge: secret values should not be used as env variables. The k8s secret should be mounted as a volume and then read from there...

File: installer/manifests/keptn/charts/control-plane/templates/_helpers.tpl [hunk 2]
```diff
@@ -137,4 +137,14 @@ lifecycle:
     fieldRef:
       fieldPath: metadata.labels['app.kubernetes.io/name']
 {{- end }}
+- name: OAUTH_CLIENT_ID
+  value: "{{ (((.Values.distributor).config).oauth).clientID }}"
+- name: OAUTH_CLIENT_SECRET
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/_helpers.tpl:142]: same comment as for the jmeter service

**Review by bacherfl — APPROVED** [review]

File: jmeter-service/chart/templates/deployment.yaml (hunk 1 above)
Comment (warber) [reply jmeter-service/chart/templates/deployment.yaml:152]: Thanks @mowies for this remark.. I think we agreed with @thisthat and @agrimmer that we will keep it like env vars for now. correct?

Comment (agrimmer) [reply jmeter-service/chart/templates/deployment.yaml:152]: Correct. Currently, we prefer to consistently use ENV vars.

**Pull request merged**